lab2.py

```diff
@@ -0,0 +1,32 @@
+from lab1 import quick_sort, merge_sort
+from timing import timing
+import random
+
+@timing
+def onsets(data, p, algorithm):
+    gen = ((el[1], el[0]) for el in enumerate(data) if el[1] > p)
+    elements = [next(gen) for i in range(3)]
+
+    if algorithm.__name__ == 'quick_sort':
+        algorithm(elements, 0, len(elements)-1)
```

Comment:
Jakby nie było tych bezsensownych dodatkowych argumentów wejściowych dla quicksorta to uniknęlibyśmy tej dodatkowej komplikacji w kodzie. Wyobraź sobie co by było, gdyby projekt był bardziej skomplikowany.